src/components/common/StaffPayment.vue

```diff
@@ -6,8 +6,8 @@
 
     <staff-payment-component
       :staffPaymentData="getStaffPayment"
-      @update:staffPaymentData="onStaffPaymentData"
-      @valid="staffPaymentFormValid = $event"
```

Comment:
I got rid of the local property and watcher, and now just call the setter directly.